manifests/server/etcd/setup.pp

```diff
@@ -56,50 +97,61 @@
         ensure => 'absent',
       }
     }
-  } else {
-    package { 'etcd':
+
+    group { $group:
       ensure => $ensure,
-      name   => $package,
+      system => true,
+      gid    => $gid,
     }
-  }
 
-  user { 'etcd':
-    ensure => $ensure,
-  }
-  group { 'etcd':
-    ensure  => $ensure,
-    members => ['etcd'],
+    user { $user:
+      ensure     => $ensure,
+      comment    => 'ETCD user',
+      gid        => $gid,
+      home       => $storage_path,
+      managehome => false,
+      shell      => (fact('os.family') ? {
+          'Debian' => '/usr/sbin/nologin',
+          default  => '/sbin/nologin',
+      }),
+      system     => true,
+      uid        => $uid,
+    }
+  } else {
+    package { $package:
+      ensure => stdlib::ensure($ensure, 'package'),
+    }
   }
 
   file {
     default:
       ensure => stdlib::ensure($ensure, 'directory');
 
     '/etc/etcd': ;
-    '/var/lib/etcd':
-      owner => 'etcd',
-      group => 'etcd';
+    $storage_path:
+      owner => $user,
+      group => $group;
   }
 
   # Use generated certs by default
   if !$k8s::server::etcd::self_signed_tls and $k8s::server::etcd::manage_certs {
-    $_dir = '/var/lib/etcd/certs'
-    $_cert_file = pick($cert_file, "${_dir}/etcd-server.pem")
-    $_key_file = pick($key_file, "${_dir}/etcd-server.key")
-    $_trusted_ca_file = pick($trusted_ca_file, "${_dir}/client-ca.pem")
-    $_client_cert_auth = pick($client_cert_auth, true)
-    $_peer_cert_file = pick($peer_cert_file, "${_dir}/etcd-peer.pem")
-    $_peer_key_file = pick($peer_key_file, "${_dir}/etcd-peer.key")
-    $_peer_trusted_ca_file = pick($peer_trusted_ca_file, "${_dir}/peer-ca.pem")
+    $_dir                   = "${storage_path}/certs"
+    $_cert_file             = pick($cert_file, "${_dir}/etcd-server.pem")
+    $_key_file              = pick($key_file, "${_dir}/etcd-server.key")
+    $_trusted_ca_file       = pick($trusted_ca_file, "${_dir}/client-ca.pem")
+    $_client_cert_auth      = pick($client_cert_auth, true)
+    $_peer_cert_file        = pick($peer_cert_file, "${_dir}/etcd-peer.pem")
+    $_peer_key_file         = pick($peer_key_file, "${_dir}/etcd-peer.key")
+    $_peer_trusted_ca_file  = pick($peer_trusted_ca_file, "${_dir}/peer-ca.pem")
     $_peer_client_cert_auth = pick($peer_client_cert_auth, true)
   } else {
-    $_cert_file = $cert_file
-    $_key_file = $key_file
-    $_trusted_ca_file = $trusted_ca_file
-    $_client_cert_auth = $client_cert_auth
-    $_peer_cert_file = $peer_cert_file
-    $_peer_key_file = $peer_key_file
-    $_peer_trusted_ca_file = $peer_trusted_ca_file
+    $_cert_file             = $cert_file
```

Comment:
I'm personally not a huge fan of this alignment because it causes a lot of noise for each new variable if it's long. That makes rebasing and cherry-picking hard. But I think we haven't decided yet what's our style guide here.

Reply:
yeah sure, i tend to to do things like that. i find it easier to read and it is better (for me) to find errors in this way. but it is just a personal thing.